components/mpas-ocean/src/shared/mpas_ocn_gm.F

```diff
@@ -680,37 +680,60 @@ subroutine ocn_GM_compute_Bolus_velocity(statePool, &
 
                BruntVaisalaFreqTopEdge = 0.5_RKIND*(BruntVaisalaFreqTop(k, cell1) + BruntVaisalaFreqTop(k, cell2))
                BruntVaisalaFreqTopEdge = max(BruntVaisalaFreqTopEdge, 0.0_RKIND)
-               tridiagB(k - 1) = -2.0_RKIND*cGMphaseSpeed(iEdge)**2/(layerThickEdge(k - 1, iEdge) &
-                                                                     *layerThickEdge(k, iEdge)) - BruntVaisalaFreqTopEdge
-               tridiagC(k - 1) = 2.0_RKIND*cGMphaseSpeed(iEdge)**2/layerThickEdge(k, iEdge) &
-                                 /(layerThickEdge(k - 1, iEdge) + layerThickEdge(k, iEdge))
-               rightHandSide(k - 1) = gmBolusKappa(iEdge)*gmKappaScaling(k,iEdge)*gravity/rho_sw &
-                                      *gradDensityConstZTopOfEdge(k,iEdge)
+               if (config_GM_gammaF_factor .eq. 0.0_RKIND) then
```

Comment:
@katsmith133 if not too much trouble it would be good to not have these if loops inside the do loop.  I'd suggest further down in the GM_init routine to have a little block of code that does something like

```
if (config_GM_gammaF_factor .eq. 0.0_RKIND) then
   gmTermBFactor = 1.0_RKIND
else
  gmTermBFactor = config_GM_gammaF_factor
endif
```
you would then put the gmTermBFactor on your tridiagB/C terms and define that variable at the top of this file.  With this you only check and if/else once instead of every point every timestep.

Let me know if this doesn't make sense.

Reply:
Yup, makes sense! I'll put that in.

Reply:
@vanroekel Ok, this new commit should fix that.